backend/kernelCI_app/views/groupedTestsView.py

```diff
@@ -0,0 +1,97 @@
+from django.http import JsonResponse, HttpResponseBadRequest
+from django.views import View
+from django.db.models import Q, TextField, Sum, Count, Case, When, F, Func, Value
+from django.db.models.functions import Concat
+from django.contrib.postgres.aggregates import ArrayAgg
+
+from kernelCI_app.models import Tests
+from kernelCI_app.serializers import GroupedTestsSerializer
+from kernelCI_app.utils import getErrorResponseBody
+
+
+class groupedTests(View):
+
+    def get(self, request):
+        commit = request.GET.get('git_commit_hash')
+        branch = request.GET.get('git_repository_branch')
+        url = request.GET.get('git_repository_url')
+        origin = request.GET.get('origin')
+        if not commit:
+            return HttpResponseBadRequest(
+                getErrorResponseBody('missing required field `git_commit_hash`')
+            )
```

Comment:
we can do something like

```
    get_required_params = ['git_commit_hash', 'git_repository_branch', 'git_repository_url' 'origin']

    def validate_required_params(self, request):
        for p in self.get_required_params:
            if not request.GET.get(p):
                return HttpResponseBadRequest(
                getErrorResponseBody(f'missing required field `{p}`')
            )
```

we can even put `validate_required_params` in utils file so we can reuse it as `validate_required_params(params, request)`